src/interpreter_funcs/voxel_boolean_union.rs

```diff
@@ -218,30 +221,77 @@ impl Func for FuncBooleanUnion {
             return Err(error);
         }
 
-        let mut voxel_cloud1 = ScalarField::from_mesh(mesh1, &voxel_dimensions, 0.0, growth_u32);
-        let mut voxel_cloud2 = ScalarField::from_mesh(mesh2, &voxel_dimensions, 0.0, growth_u32);
+        // FIXME: The growth + 1 here is only because it seems like some
+        // chopping happens. This is one place where the chopping may occur.
+        let mut voxel_cloud1 =
```

Comment:
I think the chopping you mention can be caused by using cartesian bounding boxes instead of voxel space bounding boxes. Try if it helps and maybe we don't have to hack around here.